src/app/movies.service.ts

```diff
@@ -1,11 +1,13 @@
+import { environment } from 'src/environments/environment';
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/internal/Observable';
 
 export interface PopularMovies {
-  results: Array<PopularMoviesResult>
+  results: Array<MoviesResult>
```

Comment:
Nie potrzebujesz robić osobnych intefejsów jeśli zawierają to samo a już na pewno nie są potrzebne w przypadku "PopularMovies" czy "SearchMovies". To poprostu jest Array<MovieResult> i tak to możesz zadeklarować w komponentach